app/app.py

```diff
@@ -35,52 +34,54 @@ def fix_transfer_encoding():
     """
 
     transfer_encoding = request.headers.get("Transfer-Encoding", None)
-    if transfer_encoding == u"chunked":
+    if transfer_encoding == "chunked":
         request.environ["wsgi.input_terminated"] = True
 
 
-@app.route('/')
-def hello_world():
-   return "deploy.altlab.dev is running!"
-
-
-@app.route('/<app_name>', methods=["POST"])
-def secret_assert(app_name):
-    if '..' in app_name or '/' in app_name:
-        return abort(401)
+@app.route("/<app_name>", methods=["POST"])
+def deploy_app(app_name):
+    if ".." in app_name or "/" in app_name:
+        return abort(HTTPStatus.BAD_REQUEST)
 
     body = request.get_json()
     if not isinstance(body, dict):
-        return abort(401)
+        return abort(HTTPStatus.BAD_REQUEST)
 
-    provided_secret = body.get('secret', None)
+    provided_secret = body.get("secret", None)
     expected_secret = get_app_secret(app_name)
-    if expected_secret is None:
-        return abort(401)
+    if expected_secret == "":
+        return abort(HTTPStatus.NOT_FOUND)
 
     # constant-time compare with compare_digest; it accepts ASCII strings or
     # byte string, so encode to bytes to prevent TypeError
-    if not hmac.compare_digest(provided_secret.encode("UTF-8"), expected_secret.encode('UTF-8')):
-        return abort(401)
+    if not hmac.compare_digest(
+        provided_secret.encode("ASCII"), expected_secret.encode("ASCII")
```

Comment:
Why did you change UTF-8 to ASCII? The comment explained that it was avoiding a crash with TypeError if non-ASCII input was given, but with this proposed change, non-ASCII input will crash the server with UnicodeEncodeError.

Reply:
Ah, makes sense, thanks!